src/acceptance-test/java/de/zalando/aruha/nakadi/repository/db/EventTypeDbRepositoryTest.java

```diff
@@ -0,0 +1,159 @@
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.jdbc.support.rowset.SqlRowSet;
+
+import javax.sql.DataSource;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.IsNull.notNullValue;
+
+public class EventTypeDbRepositoryTest {
+
+    private JdbcTemplate template;
+    private EventTypeRepository repository;
+    private Connection connection;
+
+    private static final String postgresqlUrl = "jdbc:postgresql://localhost:5432/local_nakadi_db";
+    private static final String username = "nakadi_app";
+    private static final String password = "nakadi";
+
+    @Before
+    public void setUp() {
+        try {
+            ObjectMapper mapper = (new NakadiConfig()).jacksonObjectMapper();
+
+            DataSource datasource = new DriverManagerDataSource(postgresqlUrl, username, password);
+            template = new JdbcTemplate(datasource);
+            repository = new EventTypeDbRepository(template, mapper);
+            connection = datasource.getConnection();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenCreateNewEventTypePersistItInTheDatabase() throws Exception {
+        EventType eventType = buildEventType();
+
+        repository.saveEventType(eventType);
+
+        final int rows = template.queryForObject("SELECT count(*) FROM zn_data.event_type", Integer.class);
+        assertThat("Number of rows should encrease", rows, equalTo(1));
+
+        SqlRowSet rs = template.queryForRowSet("SELECT et_name, et_event_type_object FROM zn_data.event_type");
+        rs.next();
+
+        assertThat("Name is persisted", rs.getString(1), equalTo("event-name"));
+
+        ObjectMapper mapper = (new NakadiConfig()).jacksonObjectMapper();
+        EventType persisted = mapper.readValue(rs.getString(2), EventType.class);
+
+        assertThat(persisted.getCategory(), equalTo(eventType.getCategory()));
+        assertThat(persisted.getName(), equalTo(eventType.getName()));
+        assertThat(persisted.getEventTypeSchema().getType(), equalTo(eventType.getEventTypeSchema().getType()));
+        assertThat(persisted.getEventTypeSchema().getSchema(), equalTo(eventType.getEventTypeSchema().getSchema()));
+    }
+
+    @Test(expected = DuplicatedEventTypeNameException.class)
+    public void whenCreateDuplicatedNamesThrowAnError() throws Exception {
+        EventType eventType = buildEventType();
+
+        repository.saveEventType(eventType);
+        repository.saveEventType(eventType);
+    }
+
+    @Test
+    public void whenEventExistsFindByNameReturnsSomething() throws NakadiException, DuplicatedEventTypeNameException, NoSuchEventTypeException {
```

Comment:
I think there's a possibility to improve this test - because it could not detect a situation that the findByName method was missing 'WHERE' clause. (at least to add several EventTypes as a bootstrap)